### PR TITLE
Feature/backlog 5984 - for documentation review only - do NOT merge

### PR DESCRIPTION
**Pull request by dcleao**
@swagner8412 please review. 
/cc @pentaho/millenniumfalcon 

It's better to focus on the "files changed tab". The relevant files are the ones with a jsdoc extension and inside the `"_doc"` folder: `"package-res/resources/web/pentaho/lang/_doc"`.

Note that file `EventSource.js` contains duplicate and outdated doclets, for now, so please ignore this one (not to confuse with `EventSource.jsdoc`)!

### CODE REVIEW

File: package-res/resources/web/pentaho/lang/_doc/Event.jsdoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ * ##### Event Cancellation
+ *
+ * Certain types of events are used to signal that an _action_ is about to execute
+ * (or that a phase of an already executing action is about to start).
+ * When the execution of the action can be canceled by the event listeners,
+ * the event is said to be _cancelable_.
+ * That characteristic is exposed by the [isCancelable]{@link pentaho.lang.Event#isCancelable} property.
+ *
+ * When an event is canceled, not only its corresponding action is canceled,
+ * but the listeners of unprocessed registrations are not notified.
```
Comment (swagner8412) [inline package-res/resources/web/pentaho/lang/_doc/Event.jsdoc]: When an event is canceled, its corresponding action is also canceled and the listeners of unprocessed registrations are not notified.

File: package-res/resources/web/pentaho/lang/_doc/Event.jsdoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ *
+ * To cancel an event, call its [cancel]{@link pentaho.lang.Event#cancel} method.
+ * To know if an event has been canceled, read the [isCanceled]{@link pentaho.lang.Event#isCanceled} property.
```
Comment (swagner8412) [inline package-res/resources/web/pentaho/lang/_doc/Event.jsdoc]: To find out if an event has been canceled, read the ......

Note: ...... means that the words that follow are fine as they already are. :)

File: package-res/resources/web/pentaho/lang/_doc/EventListener.jsdoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * An event listener function.
+ *
+ * This type can be used generically by specifying the type of event in angle brackets:
```
Comment (swagner8412) [inline package-res/resources/web/pentaho/lang/_doc/EventListener.jsdoc]: This looks good to me :)

File: package-res/resources/web/pentaho/lang/_doc/IDisposable.jsdoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * @classdesc The `IDisposable` interface represents a resource that can be
+ *   explicitly and deterministically disposed.
+ *
+ * @name pentaho.lang.IDisposable
+ * @interface
+ */
+
+/**
+ * Disposes an object if it hasn't been disposed yet.
```
Comment (swagner8412) [inline package-res/resources/web/pentaho/lang/_doc/IDisposable.jsdoc]: Disposes of an object if it hasn't been disposed of yet.

File: package-res/resources/web/pentaho/lang/_doc/IEventRegistrationHandle.jsdoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Removes the event registration, if it hasn't been removed yet.
+ *
+ * This method is an alias for [dispose]{@link pentaho.lang.IEventRegistrationHandle#dispose}.
+ *
+ * @name remove
+ * @memberOf pentaho.lang.IEventRegistrationHandle#
+ * @method
+ * @see pentaho.lang.IDisposable#dispose
+ */
```
Comment (swagner8412) [inline package-res/resources/web/pentaho/lang/_doc/IEventRegistrationHandle.jsdoc]: This one is good to go!

**Comment by dcleao** [comment]
Done its job.

**Pull request closed without merging**